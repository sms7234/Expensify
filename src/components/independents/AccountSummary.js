import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';

export const AccountSummary =({count}) => {
  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title"> Account List </h1>
        <h2 className="page-header__subtitle">Total number of accounts: <span>{count}</span></h2>
        <div className="page-header__actions">
          <Link className="button" to="/createAccount">Add Account</Link>
        </div>
      </div>
    </div>
  )
}

const mapStateToProps = (state) => {
  return {
    count: state.accounts.length
  };
};

export default connect(mapStateToProps)(AccountSummary);
