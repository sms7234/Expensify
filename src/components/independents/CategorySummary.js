import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';

export const CategorySummary = ({count}) => {

  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title"> Category List </h1>
        <h2 className="page-header__subtitle">Total number of categories: <span>{count}</span></h2>
        <div className="page-header__actions">
          <Link className="button" to="/createCategory">Add Category</Link>
        </div>
      </div>
    </div>
  )
}

const mapStateToProps = (state) => {
  return {
    count: state.categories.length
  };
};

export default connect(mapStateToProps)(CategorySummary);
