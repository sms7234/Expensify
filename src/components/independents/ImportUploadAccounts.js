import React from 'react';
import Papa from 'papaparse';
import Accordion from 'react-bootstrap/Accordion';
import Card from 'react-bootstrap/Card'

class ImportUploadAccounts extends React.Component {
  constructor(props){
    super(props);
    this.state={
      data:this.props.data,
      validation:this.props.validation,
      buttonUpload: true,
      file: null,
      error:''
    };
  };
  onFileUpload = (e)=>{
    const file = e.target.files[0];
    if (file.name.match(/\.csv$/)){
      this.setState(()=> ({
        file,
        error:'',
        buttonUpload: false
      }))
    } else {
      this.setState(()=> ({
        error: 'Please upload correct file type'
      }))
    }
  };
  onDataUpload=(e)=>{
    e.preventDefault();
    if (this.state.error==='') {
      Papa.parse(this.state.file,{
        header: true,
        dynamicTyping: true,
        complete: (results) => {
          const data = this.state.data;
          const validation = this.state.validation;
          results.data.forEach((item) => {
            item.Account = item.Account ? item.Account:'';
            item.Description = item.Description ? item.Description:'';
            data.push(item);
            validation.push(null);
          })
          this.props.onDataUpload(data, validation);
          this.setState(()=>({buttonUpload: true}));
        },
        error: (err) => {
          this.setState(()=>{error: 'Error during parse: ', err.type})
        }
      });
    }
  };
  render () {
    return (
      <div className="content-container--subHeader">
        <Accordion>
          <Card>
            <Accordion.Toggle as={Card.Header} className="accordion__header" eventKey="0">
              Upload Data
            </Accordion.Toggle>
            <Accordion.Collapse eventKey="0">
              <Card.Body>
                <form className="form" onSubmit={this.onDataUpload}>
                  {this.state.error!=='' && <h3 className="form__error">{this.state.error}</h3>}
                  <input
                    type="file"
                    name="file"
                    onChange={this.onFileUpload}
                  />
                  <div>
                    <button className="button" disabled={this.state.buttonUpload}>
                      Upload Data
                    </button>
                  </div>
                </form>
              </Card.Body>
            </Accordion.Collapse>
          </Card>
        </Accordion>
      </div>
    )
  }
}

export default ImportUploadAccounts;
