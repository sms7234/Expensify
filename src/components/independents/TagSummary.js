import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';

export const TagSummary =({count}) => {
  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title"> Tag List </h1>
        <h2 className="page-header__subtitle">Total number of tags: <span>{count}</span></h2>
        <div className="page-header__actions">
          <Link className="button" to="/createTag">Add Tag</Link>
        </div>
      </div>
    </div>
  )
}

const mapStateToProps = (state) => {
  return {
    count: state.tags.length
  };
};

export default connect(mapStateToProps)(TagSummary);
