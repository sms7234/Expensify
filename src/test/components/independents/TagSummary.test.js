import React from 'react';
import {shallow} from 'enzyme';
import {TagSummary} from '../../../components/independents/TagSummary';

test('should correctly render tag summary with no expenses',() => {
  const wrapper = shallow(<TagSummary count={0} />);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render tag summary with multiple expenses',() => {
  const wrapper = shallow(<TagSummary count={47} />);
  expect(wrapper).toMatchSnapshot();
});
