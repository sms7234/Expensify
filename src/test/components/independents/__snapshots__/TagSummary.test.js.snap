// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render tag summary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
       Tag List 
    </h1>
    <h2
      className="page-header__subtitle"
    >
      Total number of tags: 
      <span>
        47
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/createTag"
      >
        Add Tag
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render tag summary with no expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
       Tag List 
    </h1>
    <h2
      className="page-header__subtitle"
    >
      Total number of tags: 
      <span>
        0
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/createTag"
      >
        Add Tag
      </Link>
    </div>
  </div>
</div>
`;
